drivers/gpu/drm/lima/lima_gem.c

```diff
@@ -426,3 +427,81 @@ int lima_gem_wait(struct drm_file *file, u32 handle, u32 op, u64 timeout_ns)
 	drm_gem_object_unreference_unlocked(obj);
 	return ret;
 }
+
+static struct lima_bo *lima_gem_alloc_object_dmabuf(struct drm_device *dev,
+		size_t size)
+{
+	struct lima_bo *bo;
+	int err;
+
+	size = roundup(size, PAGE_SIZE);
+
+	bo = kzalloc(sizeof(*bo), GFP_KERNEL);
+	if (!bo)
+		return ERR_PTR(-ENOMEM);
+
+	mutex_init(&bo->lock);
+	INIT_LIST_HEAD(&bo->va);
+	reservation_object_init(&bo->resv);
+
+	err = drm_gem_object_init(dev, &bo->gem, size);
+	if (err)
+		goto err_out0;
+
+	return bo;
+
+err_out0:
+	kfree(bo);
+	return ERR_PTR(err);
+}
+
+struct drm_gem_object *lima_gem_prime_import(struct drm_device *dev,
```

Comment:
we can just use the drm_gem_prime_import then
bo->dma_addr = sg_dma_address(sgt->sgl);
in this function with the above gem_bo create change